CareKit/CareKit/Shared/Contact/DetailedContactView.swift

```diff
@@ -0,0 +1,138 @@
+///     | +---------------------------------------------------+ |
+///     | |  [title]                                          | |
+///     | |  [detail]                                         | |
+///     | |                                                   | |
+///     | +---------------------------------------------------+ |
+///     +-------------------------------------------------------+
+///
+@available(iOS 14.0, watchOS 7.0, *)
+public struct DetailedContactView<Header: View, Footer: View>: View {
+    private typealias ContactView = SynchronizedContactView<OCKDetailedContactController, CareKitUI.DetailedContactView<Header, Footer>>
+
+    private let contactView: ContactView
+    
+    public var body: some View {
+        contactView
+    }
+
+    private init(contactView: ContactView) {
+        self.contactView = contactView
+    }
+    
+    public init(contactID: String, storeManager: OCKSynchronizedStoreManager, content: @escaping (_ controller: OCKDetailedContactController) -> CareKitUI.DetailedContactView<Header, Footer>) {
+        contactView = .init(controller: .init(storeManager: storeManager), query: .contactID(contactID), content: content)
+    }
+    
+    public init(contact: OCKAnyContact, contactQuery: OCKContactQuery, storeManager: OCKSynchronizedStoreManager,
+                content: @escaping (_ controller: OCKDetailedContactController) -> CareKitUI.DetailedContactView<Header, Footer>) {
+        contactView = .init(controller: .init(storeManager: storeManager), query: .init(.contact(contact)), content: content)
+    }
+    
+    public init(controller: OCKDetailedContactController, content: @escaping (_ controller: OCKDetailedContactController) -> CareKitUI.DetailedContactView<Header, Footer>) {
+        contactView = .init(controller: controller, content: content)
+    }
+    
+    public func onError(_ perform: @escaping (Error) -> Void) -> DetailedContactView<Header, Footer> {
+        .init(contactView: .init(copying: contactView, settingErrorHandler: perform))
+    }
+}
+
+@available(iOS 14.0, watchOS 7.0, *)
+public extension DetailedContactView where Header == _DetailedContactViewHeader, Footer == _DetailedContactViewFooter {
+    init(contactID: String, storeManager: OCKSynchronizedStoreManager) {
+        self.init(contactID: contactID, storeManager: storeManager) {
+            .init(viewModel: $0.viewModel)
+        }
+    }
+    
+    init(contact: OCKAnyContact,contactQuery: OCKContactQuery, storeManager: OCKSynchronizedStoreManager) {
+        self.init(contact: contact, contactQuery: contactQuery, storeManager: storeManager) {
+            .init(viewModel: $0.viewModel)
+        }
+    }
+    
+    init(controller: OCKDetailedContactController) {
+        contactView = .init(controller: controller) {
+            .init(viewModel: $0.viewModel)
+        }
+    }
+}
+
+// CODE REVIEW: Let's add actions - they are nil right now, see UIKit version
```

Comment:
The actions should be closure properties on the `ContactViewModel` that starts a few lines below this.

Reply:
done